src/boot/split.rs

```diff
@@ -36,156 +37,414 @@ fn executable_permissions() -> Option<Permissions> {
+    // the compressed size of the zipped up directory. We want the uncompressed size of the zipped
+    // up directory here as the proper fair warning on a split, which will extract the directory
+    // loose.
+    print!(
+        "{path} {size} {file_type}",
+        path = base.as_ref().join(&file.name).display(),
+        size = zip_archive.decompressed_size().unwrap_or(file.size as u128),
+        file_type = file_type(file)
+    );
+    if let Some(key) = &file.key {
+        print!(" ({key})");
+    }
+    println!();
+}
+
+fn extract_to<P: AsRef<Path>, R: Read>(base: P, file: &File, reader: &mut R) -> Result<u64, Exit> {
+    let dst = ensure_parent_dir(base.as_ref(), file)?;
+    let mut out = std::fs::OpenOptions::new()
         .write(true)
         .create_new(true)
-        .open(base.join("lift.json"))
+        .open(&dst)
         .map_err(|e| {
-            Code::FAILURE.with_message(format!("Failed to open lift manifest for writing: {e}"))
+            Code::FAILURE.with_message(format!(
+                "Failed to open {dst} for extraction: {e}",
+                dst = dst.display()
+            ))
         })?;
+    std::io::copy(reader, &mut out).map_err(|e| {
+        Code::FAILURE.with_message(format!(
+            "Failed to extract {file} to {dst}: {e}",
+            file = file.name,
+            dst = dst.display()
+        ))
+    })
+}
+
+struct EmbeddedZipReader<'a, S> {
+    scie: &'a mut S,
+    start: u64,
+    length: u64,
+    offset: u64,
+}
+
+impl<'a, S> EmbeddedZipReader<'a, S> {
+    pub fn new(scie: &'a mut S, start: u64, length: u64) -> Self {
+        Self {
+            scie,
+            start,
+            length,
+            offset: 0,
+        }
+    }
+}
+
+impl<R: Read + Debug> Read for EmbeddedZipReader<'_, R> {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        let written = self.scie.take(self.length - self.offset).read(buf)?;
+        self.offset += written as u64;
+        Ok(written)
+    }
+}
+
+impl<S: Seek + Debug> Seek for EmbeddedZipReader<'_, S> {
+    fn seek(&mut self, pos: SeekFrom) -> io::Result<u64> {
+        let offset = match pos {
+            SeekFrom::Start(offset) => Some(offset),
+            SeekFrom::End(offset) => self.length.checked_add_signed(offset),
+            SeekFrom::Current(offset) => self.offset.checked_add_signed(offset),
+        };
+        match offset {
+            None => Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "invalid seek to a negative or overflowing position",
+            )),
+            Some(offset) => {
+                let clamped_offset = cmp::min(self.length, offset);
+                let new_inner_offset = self
+                    .scie
+                    .seek(SeekFrom::Start(self.start + clamped_offset))?;
+                self.offset = new_inner_offset - self.start;
+                Ok(self.offset)
+            }
+        }
+    }
+}
```

Comment:
Much like in the installer improvement above, this allows reading entries out of a scie-tote while still embedded in the scie. No tempfile need be resorted to any longer.